### DISCUSSION TITLE
Huge memory consumption when converting jpg with clipping path to transparent png

### DISCUSSION THREAD

**Discussion by kurkel** [ImageMagick/ImageMagick#3706] [Help] [answered]
So I have a JPG file with fairly complex clipping path including islands. Still the file itself is quite small.
```
identify file.jpg         
file.jpg JPEG 2054x900 2054x900+0+0 8-bit sRGB 1.06499MiB 0.000u 0:00.005
```

When running `magick file.jpg -alpha opaque -clip -alpha transparent +clip file.png` on my MBP ImageMagick peaks at 7 GB of memory. The same happens even with the following config I've set:
```
magick -list resource
Resource limits:
  Width: 461.169PP
  Height: 461.169PP
  Area: 512000000P
  List length: unlimited
  Memory: 512000000B
  Map: 512000000B
  Disk: unlimited
  File: 192
  Thread: 12
  Throttle: 0
  Time: unlimited 
```

This is problematic as we have an automated Heroku pipeline which processes these third party provided images into transparent background assets and the dyno in question only has 1 gig of memory.

Any idea if this is normal behaviour and I should just find an instance with enough memory? Could the ignorance of memory limits be due to some delegate being the culprit?

Here's my version info (Mac OS Catalina 10.15.7):
```
magick -version
Version: ImageMagick 7.0.11-9 Q16 x86_64 2021-04-24 https://imagemagick.org
```
And the heroku dyno's ( Heroku-18 - Ubuntu 18.04 with this buildpack: https://github.com/yespark/heroku-imagemagick-buildpack)
```
magick -version
Version: ImageMagick 7.0.11-12 Q16 x86_64 2021-05-09 https://imagemagick.org
```

**Comment by snibgo** [comment]
The clip path may be weird. Can you paste a link to the JPG file?

If you can't, then please paste the text output from:
```
magick file.jpg -verbose info:
```

**Reply by kurkel** [reply]
I'm unsure if I'm allowed to share the file, I will know tomorrow. For the mean time here is the info:
[info.txt](https://github.com/ImageMagick/ImageMagick/files/6509833/info.txt)

The clip path sure looks long...

**Comment by snibgo** [comment, marked as answer]
Yes, the clipping path is huge, with about 66000 curves. It is also weird. For example, the second line is:
```
C 639.652 876.976 639.652 876.632 639.652 876.289
```
This defines a curve with three control points. Those points have identical x-coordinates, and almost identical y-coordinates. I suppose this "curve" has at most two pixels.

The next curve is almost identical to the first curve. And so it goes on. And on. And on. Ths generates a polygon, defined by about 66000 curves, which is then rasterized. I suspect this explains the massive memory requirement.

Perhaps the software that generated that clippath might generate a simpler version.

**Reply by kurkel** [reply]
Yeah that seems to be the source of the problem, thanks mate!
